File: backend/src/main/java/de/interaapps/pastefy/controller/publicarea/PublicUserController.java
package de.interaapps.pastefy.controller.publicarea;

import de.interaapps.pastefy.controller.HttpController;
import de.interaapps.pastefy.exceptions.NotFoundException;
import de.interaapps.pastefy.model.database.User;
import de.interaapps.pastefy.model.responses.user.PublicUserResponse;
import org.javawebstack.http.router.router.annotation.PathPrefix;
import org.javawebstack.http.router.router.annotation.With;
import org.javawebstack.http.router.router.annotation.params.Path;
import org.javawebstack.http.router.router.annotation.verbs.Get;

@With("public-pastes-endpoint")
@PathPrefix("/api/v2/public/user")
public class PublicUserController extends HttpController {
    @Get("/{name}")
    public PublicUserResponse getUser(@Path("name") String name) {
        User user = User.getByName(name);
        if (user == null) throw new NotFoundException();
        return new PublicUserResponse(user);
    }
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/controller/public/PublicUserController.kt
package de.interaapps.pastefy.controller.public

import de.interaapps.pastefy.auth.annotations.PublicPastesEnabled
import de.interaapps.pastefy.dto.user.PublicUserDto
import de.interaapps.pastefy.exceptions.NotFoundException
import de.interaapps.pastefy.service.UserService
import de.interaapps.pastefy.service.toPublicDto
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.PathVariable

@RestController
@RequestMapping("/api/v2/public/user")
@PublicPastesEnabled
class PublicUserController(
    private val users: UserService,
) {
    @GetMapping("/{name}")
    fun getUser(@PathVariable name: String): PublicUserDto =
        users.getByName(name)?.toPublicDto() ?: throw NotFoundException()
}
